pkg/util/node/node.go

```diff
@@ -103,20 +103,11 @@ func GetNodeHostIPs(node *v1.Node) ([]net.IP, error) {
 	return nodeIPs, nil
 }
 
-// GetNodeHostIP returns the provided node's "primary" IP; see GetNodeHostIPs for more details
-func GetNodeHostIP(node *v1.Node) (net.IP, error) {
-	ips, err := GetNodeHostIPs(node)
-	if err != nil {
-		return nil, err
-	}
-	// GetNodeHostIPs always returns at least one IP if it didn't return an error
-	return ips[0], nil
-}
-
-// GetNodeIP returns an IP (as with GetNodeHostIP) for the node with the provided name.
-// If required, it will wait for the node to be created.
-func GetNodeIP(client clientset.Interface, name string) net.IP {
-	var nodeIP net.IP
+// GetNodeIPs returns IPs for the node with the provided name.  If
+// required, it will wait for the node to be created.
+func GetNodeIPs(ctx context.Context, client clientset.Interface, name string) []net.IP {
```

Comment:
As a "library" function, it would make much more sense for this to return `([]net.IP, error)` rather than for it to take a `context.Context` and log errors itself.

(Yes, that's not how it worked before, but it was bad before.)